tests/by-util/test_cp.rs

```diff
@@ -4068,6 +4068,213 @@ fn test_cp_dest_no_permissions() {
         .stderr_contains("denied");
 }
 
+/// Regression test for macOS readonly file behavior (issue #5257, PR #5261)
+#[test]
+fn test_cp_readonly_dest_regression() {
+    let ts = TestScenario::new(util_name!());
+    let at = &ts.fixtures;
+
+    // Batch file operations to reduce I/O overhead
+    at.write("source.txt", "source content");
+    at.write("readonly_dest.txt", "original content");
+    at.set_readonly("readonly_dest.txt");
+
+    ts.ucmd()
+        .args(&["source.txt", "readonly_dest.txt"])
+        .fails()
+        .stderr_contains("readonly_dest.txt")
+        .stderr_contains("denied");
+
+    assert_eq!(at.read("readonly_dest.txt"), "original content");
+}
+
+/// Test readonly destination behavior with --force flag (should succeed)
+#[cfg(not(windows))]
+#[test]
+fn test_cp_readonly_dest_with_force() {
+    let ts = TestScenario::new(util_name!());
+    let at = &ts.fixtures;
+
+    // Use consistent file operations and batch setup
+    at.write("source.txt", "source content");
+    at.write("readonly_dest.txt", "original content");
+    at.set_readonly("readonly_dest.txt");
+
+    ts.ucmd()
+        .args(&["--force", "source.txt", "readonly_dest.txt"])
+        .succeeds();
+
+    assert_eq!(at.read("readonly_dest.txt"), "source content");
+}
```

Comment:
This test is more or less the same as `test_cp_arg_force` on line 941. I would keep this test as it is cleaner and remove the other one.